.github/workflows/ci.yml

```diff
@@ -1,38 +1,39 @@
-name: CI
+name: CI/CD
+
 on:
-  pull_request:
-    branches: ['**']
   push:
-    branches: ['**']
-    tags: [v*]
+  workflow_dispatch:
+
 jobs:
   ci:
-    # run on external PRs, but not on internal PRs since those will be run by push to branch
-    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
-    runs-on: ubuntu-20.04
+    name: Test and Push
+    runs-on: ubuntu-22.04
+
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-    - name: Set up JDK 11
-      uses: actions/setup-java@v1
-      with:
-        java-version: 11
-    - name: Cache sbt
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.sbt
-          ~/.ivy2/cache
-          ~/.coursier
-        key: sbt-cache-${{ runner.os }}-${{ hashFiles('project/build.properties') }}
-    - name: Compile
-      run: sbt -v compile
-    - name: Test
-      run: sbt -v test
-    - name: Cleanup
-      run: |
-        rm -rf "$HOME/.ivy2/local" || true
-        find $HOME/.ivy2/cache                       -name "ivydata-*.properties" -delete || true
-        find $HOME/.ivy2/cache                       -name "*-LM-SNAPSHOT*"       -delete || true
-        find $HOME/.cache/coursier/v1                -name "ivydata-*.properties" -delete || true
-        find $HOME/.sbt                              -name "*.lock"               -delete || true
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: sbt
+
+      - name: Display versions
+        run: |
+          java -version
+          sbt --version
+
+      - name: Launch tests
+        working-directory: .
+        run: sbt -v test
+
+      - name: Push to Repository
+        if: github.ref == 'refs/heads/main'
+        working-directory: .
+        env:
+          NEXUS_CREDENTIALS: ${{ secrets.NEXUS_CREDENTIALS }}
+        run: |
+          echo "$NEXUS_CREDENTIALS" > ~/.sbt/.credentials_sml_nexus
+          sbt "project tradingHelloServiceApi" publish
```

Comment:
The idea of "not publishing the service artifact" was rather about putting `publishArtifact := false` in the service, but keeping the `publish` call for root project. This way this `ci.yml` file can stay universal.

Reply:
Right! sorry for misunderstanding - I have restored it and added `publishArtifact := false` to the `tradingHelloService` settings, but I have to add `build.sbt` also, as without it, sbt throw me an error:
'''
[error] (tradingHelloService / publishConfiguration) Repository for publishing is not specified.
'''
it is strange, isn't it?